action.yml

```
@@ -0,0 +1,232 @@
+          exit 1
+        fi
+        if ! [[ "$ENDTOEND_CONFIG" =~ ^[a-zA-Z0-9_\./\-]*$ ]]; then
+          echo "Invalid input for endtoend_config"
+          exit 1
+        fi
+
+    - name: Run PHPUnit
+      # input booleans are converted to strings
+      # https://docs.github.com/en/actions/learn-github-actions/expressions#functions
+      if: ${{ inputs.phpunit == 'true' }}
+      shell: bash
+      env:
+        PHPUNIT_SUITE: ${{ inputs.phpunit_suite }}
+      run: |
+        FAIL_ON_WARNING=
+        if [ "${{ inputs.phpunit_fail_on_warning }}" == "true" ]; then
+          FAIL_ON_WARNING='--fail-on-warning'
+        fi
+        echo "Running PHPUnit with options: $FAIL_ON_WARNING"
+        if [ "$PHPUNIT_SUITE" == "all" ] || [ "$PHPUNIT_SUITE" == "" ]; then
+          vendor/bin/phpunit --verbose --colors=always $FAIL_ON_WARNING
+        else
+          vendor/bin/phpunit --verbose --colors=always --testsuite "$PHPUNIT_SUITE" $FAIL_ON_WARNING
+        fi
+        echo "Passed"
+
+    - name: Setup chrome and chromedriver
+      if: ${{ inputs.endtoend == 'true' }}
+      shell: bash
+      run: |
+        echo "Default versions of google-chrome and chromedriver"
+        GCVR=$(google-chrome --version)
+        CDVR=$(chromedriver --version)
+        echo "$GCVR"
+        echo "$CDVR"
+        # Example version number is 101.0.4951.64
+        [[ "$GCVR" =~ ([0-9]+\.[0-9]+\.[0-9]+)\.[0-9]+ ]]
+        GCV="${BASH_REMATCH[1]}"
+        [[ "$CDVR" =~ ([0-9]+\.[0-9]+\.[0-9]+)\.[0-9]+ ]]
+        CDV="${BASH_REMATCH[1]}"
+        # Reinstall if a.b.c versions do not match, though allow a different .d version
+        if [ "$GCV" != "$CDV" ]; then
+          WGC=$(which google-chrome)
+          echo "google-chrome and chromedriver versions do not match, reinstalling"
+          sudo apt remove -y --purge google-chrome-stable || true
+          # Note that on ubuntu 20.04 and later, these will be installed via a snap and there seems
+          # too be a permission error with using snap on github actions, but it does not seem to matter
+          # mkdir: cannot create directory '/run/user/1001': Permission denied
+          sudo apt install -y chromium-browser chromium-chromedriver
+          echo "Updated versions of chromium-browser and chromedriver"
+          sudo ln -s $(which chromium-browser) "$WGC"
+          google-chrome --version
+          chromedriver --version
+        else
+          echo "Default versions match, continuing"
+        fi
+
+    - name: Run end-to-end tests
+      if: ${{ inputs.endtoend == 'true' }}
+      shell: bash
+      env:
+        ENDTOEND_SUITE: ${{ inputs.endtoend_suite }}
+        ENDTOEND_CONFIG: ${{ inputs.endtoend_config }}
+      run: |
+        echo "Running behat"
+        BEHAT_CONFIG="behat.yml"
+        if [ "$ENDTOEND_CONFIG" != "" ]; then
+          BEHAT_CONFIG="$ENDTOEND_CONFIG"
+        fi
+        if [ ! -f "$BEHAT_CONFIG" ]; then
+          echo "behat.yml config missing"
```

This might not be `behat.yml` if `ENDTOEND_CONFIG` sets some other file name.
```suggestion
          echo "$BEHAT_CONFIG config file missing"
```

---

Updated